### PR TITLE
Fix insert-adjacent error

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @nox

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11832)

### CODE REVIEW

File: tests/wpt/web-platform-tests/domparsing/insert-adjacent.html [hunk 1]
```diff
@@ -31,6 +31,7 @@
     el.insertAdjacentHTML(position, html);
     var heading = document.createElement('h3');
     heading.innerHTML = position;
+    /* assert_equals(el.parentElement.innerHTML, "fuck"); */
```
Comment (nox) [inline tests/wpt/web-platform-tests/domparsing/insert-adjacent.html]: What?

**Comment by jdm** [comment]
```
  ▶ Unexpected subtest result in /domparsing/insert_adjacent_html.html:
  └ PASS [expected FAIL] beforeBegin content without next sibling

  ▶ Unexpected subtest result in /domparsing/insert_adjacent_html.html:
  └ PASS [expected FAIL] afterend content without next sibling

  ▶ Unexpected subtest result in /domparsing/insert_adjacent_html.html:
  └ PASS [expected FAIL] beforeBegin content again, with next sibling

  ▶ Unexpected subtest result in /domparsing/insert_adjacent_html.html:
  └ PASS [expected FAIL] afterend content again, with next sibling

  ▶ Unexpected subtest result in /domparsing/insert_adjacent_html.html:
  └ PASS [expected FAIL] beforeBegin child node not in tree but has parent

  ▶ Unexpected subtest result in /domparsing/insert_adjacent_html.html:
  └ PASS [expected FAIL] afterend child node not in tree but has parent

  ▶ Unexpected subtest result in /domparsing/insert_adjacent_html.html:
  └ PASS [expected FAIL] beforeBegin content2 without next sibling

  ▶ Unexpected subtest result in /domparsing/insert_adjacent_html.html:
  └ PASS [expected FAIL] afterend content2 without next sibling

  ▶ Unexpected subtest result in /domparsing/insert_adjacent_html.html:
  └ PASS [expected FAIL] beforeBegin content2 test again, now that there's a next sibling

  ▶ Unexpected subtest result in /domparsing/insert_adjacent_html.html:
  └ PASS [expected FAIL] afterend content2 test again, now that there's a next sibling

  ▶ Unexpected subtest result in /domparsing/insert_adjacent_html.xhtml:
  └ PASS [expected FAIL] beforeBegin content without next sibling

  ▶ Unexpected subtest result in /domparsing/insert_adjacent_html.xhtml:
  └ PASS [expected FAIL] afterend content without next sibling

  ▶ Unexpected subtest result in /domparsing/insert_adjacent_html.xhtml:
  └ PASS [expected FAIL] beforeBegin content again, with next sibling

  ▶ Unexpected subtest result in /domparsing/insert_adjacent_html.xhtml:
  └ PASS [expected FAIL] afterend content again, with next sibling

  ▶ Unexpected subtest result in /domparsing/insert_adjacent_html.xhtml:
  └ PASS [expected FAIL] beforeBegin child node not in tree but has parent

  ▶ Unexpected subtest result in /domparsing/insert_adjacent_html.xhtml:
  └ PASS [expected FAIL] afterend child node not in tree but has parent

  ▶ Unexpected subtest result in /domparsing/insert_adjacent_html.xhtml:
  └ PASS [expected FAIL] beforeBegin content2 without next sibling

  ▶ Unexpected subtest result in /domparsing/insert_adjacent_html.xhtml:
  └ PASS [expected FAIL] afterend content2 without next sibling

  ▶ Unexpected subtest result in /domparsing/insert_adjacent_html.xhtml:
  └ PASS [expected FAIL] beforeBegin content2 test again, now that there's a next sibling

  ▶ Unexpected subtest result in /domparsing/insert_adjacent_html.xhtml:
  └ PASS [expected FAIL] afterend content2 test again, now that there's a next sibling
```

**Comment by ConnorGBrewster** [comment]
@bors-servo retry
 * #11693

**Pull request merged**